### DISCUSSION TITLE
Execute a script at container start

### DISCUSSION THREAD

**Discussion by nicoska84** [tteck/Proxmox#2655] [Q&A] [answered]
I know that maybe is a bit off topic, but I wasn't able to figure out within Proxmox support forum, so maybe here someone can help me.

I'm trying to execute a bash script `/root/vpnlogin.sh` when a lxc container starts. I tried CRON, systemd, but nothing seems to work.

If I execute it manually, `/root/vpnlogin.sh &` it works without any issue.

Thank you for your help
Nico

**Comment by tteck** [comment, marked as answer]
Ensure that `/root/vpnlogin.sh` first line is `#!/usr/bin/env bash`

Make `/root/vpnlogin.sh` executable
```
chmod +x /root/vpnlogin.sh
```
Create a service
```
cat <<EOF >/etc/systemd/system/vpnlogin.service
[Unit]
Description=vpnlogin
After=network.target

[Service]
Type=simple
ExecStart=/root/vpnlogin.sh

[Install]
WantedBy=multi-user.target
EOF
```
Enable service
```
systemctl enable -q vpnlogin.service
```
Reboot container

**Reply by nicoska84** [reply]
It don't work.
Here is the service:

```
`root@qbittorrent:~# cat /etc/systemd/system/vpnlogin.service 
[Unit]
Description=vpnlogin
After=network.target

[Service]
Type=simple
ExecStart=/root/vpnlogin.sh

[Install]
WantedBy=multi-user.target
root@qbittorrent:~# `

```
Here is the script:

```
`root@qbittorrent:~# cat vpnlogin.sh 
#!/usr/bin/env bash
openvpn --config /etc/openvpn/client/shakenmountain.ovpn &
root@qbittorrent:~# `
```

Here is the script directory and permission:

```
`root@qbittorrent:~# ls -all
-rwxr-xr-x  1 root root       79 Mar 14 09:57 vpnlogin.sh
`
```

What I'm doing wrong?

**Reply by tteck** [reply]
Try removing ` &`, it will still run in the background

Check `systemctl status vpnlogin.service` to see what it's doing.

**Comment by nicoska84** [comment]
> Try removing ` &`, it will still run in the background
> 
> Check `systemctl status vpnlogin.service` to see what it's doing.

Without the ending `&` it's working!

Thank you man! Always helpful :)
